From 1e89467e0639821e03218f87cb84616b294cf275 Mon Sep 17 00:00:00 2001
From: Bing Jiang <bing.jiang@amlogic.com>
Date: Sat, 31 Oct 2020 13:05:48 +0800
Subject: [PATCH] dts: enable absystem for ah212 [1/1]

PD#SWPL-35835

Problem:
  enable absystem for ah212

Solution:
  enable absystem for ah212

Verify:
  SC2_AH212

Change-Id: Ie8173ee7fba57ba959ba35d30ec01809f35b4762
Signed-off-by: Bing Jiang <bing.jiang@amlogic.com>
Signed-off-by: Jiacai.Liu <jiacai.liu@amlogic.com>
---
 .../amlogic/partition_mbox_normal_P_rdk.dtsi  | 128 ++++++++++++++++++
 .../sc2_s905x4_ah212_drm_buildroot.dts        |  99 +-------------
 2 files changed, 130 insertions(+), 97 deletions(-)
 create mode 100644 arch/arm/boot/dts/amlogic/partition_mbox_normal_P_rdk.dtsi

diff --git a/arch/arm/boot/dts/amlogic/partition_mbox_normal_P_rdk.dtsi b/arch/arm/boot/dts/amlogic/partition_mbox_normal_P_rdk.dtsi
new file mode 100644
index 000000000000..bde4e68354ec
--- /dev/null
+++ b/arch/arm/boot/dts/amlogic/partition_mbox_normal_P_rdk.dtsi
@@ -0,0 +1,128 @@
+/*
+ * Amlogic partition set for normal
+ *
+ * Copyright (c) 2017-2017 Amlogic Ltd
+ *
+ * This file is licensed under a dual GPLv2 or BSD license.
+ *
+ */
+
+/ {
+	partitions: partitions{
+				parts = <19>;
+				part-0 = <&dtbbak>;
+				part-1 = <&bootloaderbak>;
+				part-2 = <&recovery>;
+				part-3 = <&logo>;
+				part-4 = <&misc>;
+				part-5 = <&cri_data>;
+				part-6 = <&rsv>;
+				part-7 = <&param>;
+				part-8 = <&tee>;
+				part-9 = <&factory>;
+				part-10 = <&oem>;
+				part-11 = <&vbmeta_a>;
+				part-12 = <&boot_a>;
+				part-13 = <&system_a>;
+				part-14 = <&vbmeta_b>;
+				part-15 = <&boot_b>;
+				part-16 = <&system_b>;
+				part-17 = <&cache>;
+				part-18 = <&data>;
+		dtbbak:dtbbak{
+				   pname = "dtbbak";
+				   size = <0x0 0x100000>;
+				   mask = <1>;
+			   };
+		bootloaderbak:bootloaderbak{
+						  pname = "bootloaderbak";
+						  size = <0x0 0x200000>;
+						  mask = <1>;
+					  };
+		recovery:recovery{
+					 pname = "recovery";
+					 size = <0x0 0x4000000>;
+					 mask = <1>;
+				 };
+		logo:logo{
+				 pname = "logo";
+				 size = <0x0 0x800000>;
+				 mask = <1>;
+			 };
+		misc:misc{
+				 pname = "misc";
+				 size = <0x0 0x200000>;
+				 mask = <1>;
+			 };
+		cri_data:cri_data {
+					 pname = "cri_data";
+					 size = <0x0 0x400000>;
+					 mask = <2>;
+				 };
+		rsv:rsv{
+				pname = "rsv";
+				size = <0x0 0x1000000>;
+				mask = <1>;
+			};
+		param:param{
+				  pname = "param";
+				  size = <0x0 0x1000000>;
+				  mask = <2>;
+			  };
+		tee:tee{
+				pname = "tee";
+				size = <0x0 0x2000000>;
+				mask = <1>;
+			};
+		factory:factory{
+				pname = "factory";
+				size = <0x0 0x800000>;
+				mask = <0x11>;
+			};
+		oem:oem{
+				pname = "oem";
+				size = <0x0 0x800000>;
+				mask = <1>;
+			};
+		vbmeta_a:vbmeta_a{
+					 pname = "vbmeta_a";
+					 size = <0x0 0x200000>;
+					 mask = <1>;
+				 };
+		boot_a:boot_a {
+				   pname = "boot_a";
+				   size = <0x0 0x1000000>;
+				   mask = <1>;
+			   };
+		system_a:system_a {
+					 pname = "system_a";
+					 size = <0x0 0x96000000>;
+					 mask = <1>;
+				 };
+		vbmeta_b:vbmeta_b {
+					 pname = "vbmeta_b";
+					 size = <0x0 0x200000>;
+					 mask = <1>;
+				 };
+		boot_b:boot_b {
+				   pname = "boot_b";
+				   size = <0x0 0x1000000>;
+				   mask = <1>;
+			   };
+		system_b:system_b {
+					 pname = "system_b";
+					 size = <0x0 0x96000000>;
+					 mask = <1>;
+				 };
+		cache:cache {
+				  pname = "cache";
+				  size = <0x0 0xC800000>;
+				  mask = <2>;
+			  };
+		data:data {
+		 pname = "data";
+		 size = <0xffffffff 0xffffffff>;
+		 mask = <4>;
+	 };
+			};
+};/* end of / */
diff --git a/arch/arm/boot/dts/amlogic/sc2_s905x4_ah212_drm_buildroot.dts b/arch/arm/boot/dts/amlogic/sc2_s905x4_ah212_drm_buildroot.dts
index cba5d6653f20..f6f45e4d1624 100644
--- a/arch/arm/boot/dts/amlogic/sc2_s905x4_ah212_drm_buildroot.dts
+++ b/arch/arm/boot/dts/amlogic/sc2_s905x4_ah212_drm_buildroot.dts
@@ -19,7 +19,7 @@
 
 #include "mesonsc2.dtsi"
 #include "mesonsc2_drm.dtsi"
-//#include "partition_mbox_normal_dynamic.dtsi"
+#include "partition_mbox_normal_P_rdk.dtsi"
 //#include "mesonsm1_skt-panel.dtsi"
 
 / {
@@ -90,7 +90,7 @@
 		ion_cma_reserved:linux,ion-dev {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x8000000>;
+			size = <0xC800000>;
 			alignment = <0x400000>;
 			alloc-ranges = <0x30000000 0x50000000>;
 			no-kernel-map;
@@ -280,101 +280,6 @@
 		status = "okay";
 	};
 
-	partitions: partitions{
-		parts = <15>;
-		part-0 = <&logo>;
-		part-1 = <&recovery>;
-		part-2 = <&misc>;
-		part-3 = <&dto>;
-		part-4 = <&cri_data>;
-		part-5 = <&param>;
-		part-6 = <&boot>;
-		part-7 = <&rsv>;
-		part-8 = <&tee>;
-		part-9 = <&factory>;
-		part-10 = <&vendor>;
-		part-11 = <&odm>;
-		part-12 = <&system>;
-		part-13 = <&cache>;
-		part-14 = <&data>;
-
-		logo:logo{
-			pname = "logo";
-			size = <0x0 0x800000>;
-			mask = <1>;
-		};
-		recovery:recovery{
-			pname = "recovery";
-			size = <0x0 0x1800000>;
-			mask = <1>;
-		};
-		misc:misc{
-			pname = "misc";
-			size = <0x0 0x800000>;
-			mask = <1>;
-		};
-		dto:dto{
-			pname = "dto";
-			size = <0x0 0x800000>;
-			mask = <1>;
-		};
-		cri_data:cri_data{
-			pname = "cri_data";
-			size = <0x0 0x800000>;
-			mask = <2>;
-		};
-		rsv:rsv{
-			pname = "rsv";
-			size = <0x0 0x1000000>;
-			mask = <1>;
-		};
-		param:param{
-			pname = "param";
-			size = <0x0 0x1000000>;
-			mask = <2>;
-		};
-		boot:boot{
-			pname = "boot";
-			size = <0x0 0x1000000>;
-			mask = <1>;
-		};
-		tee:tee{
-			pname = "tee";
-			size = <0x0 0x2000000>;
-			mask = <1>;
-		};
-		factory:factory{
-			pname = "factory";
-			size = <0x0 0x800000>;
-			mask = <0x11>;
-		};
-		vendor:vendor{
-			pname = "vendor";
-			size = <0x0 0x20000000>;
-			mask = <1>;
-		};
-		odm:odm{
-			pname = "odm";
-			size = <0x0 0x10000000>;
-			mask = <1>;
-		};
-		system:system{
-			pname = "system";
-			size = <0x0 0x80000000>;
-			mask = <1>;
-		};
-		cache:cache{
-			pname = "cache";
-			size = <0x0 0x46000000>;
-			mask = <2>;
-		};
-		data:data{
-			pname = "data";
-			size = <0xffffffff 0xffffffff>;
-			mask = <4>;
-		};
-	};
-
 	ppmgr {
 		compatible = "amlogic, ppmgr";
 		memory-region = <&ppmgr_reserved>;
-- 
2.31.1

